src/modules/manaFiatGateway/sagas.ts

```diff
@@ -44,23 +50,31 @@ export function createManaFiatGatewaysSaga(config: ManaFiatGatewaySagasConfig) {
 
 function* handleOpenFiatGateway(
   config: ManaFiatGatewaySagasConfig,
-  action: OpenManaFiatGatewayAction
+  action: OpenManaFiatGatewayRequestAction
 ) {
   const { network, gateway } = action.payload
   const { transak: transakConfig, moonPay: moonPayConfig } = config
-  const address: string = yield select(getAddress)
 
-  switch (gateway) {
-    case NetworkGatewayType.TRANSAK:
-      const transak = new Transak(transakConfig, address, network)
-      transak.openWidget(network)
-      break
-    case NetworkGatewayType.MOON_PAY:
-      const widgetUrl = new MoonPay(moonPayConfig).widgetUrl(address, network)
-      window.open(widgetUrl, '_blank', 'noopener,noreferrer')
-      break
-    default:
-      break
+  try {
+    const address: string = yield select(getAddress)
+
+    switch (gateway) {
+      case NetworkGatewayType.TRANSAK:
+        const transak = new Transak(transakConfig, address, network)
```

Comment:
```suggestion
        const address: string = yield select(getAddress)
        const transak = new Transak(transakConfig, address, network)
```
I'd move the address selector to the `transak` case now that the moonpay one won't use it anymore, wdyt?